reactive/containerd.py

```diff
@@ -378,11 +414,14 @@ def config_changed():
         context
     )
 
+    DB.set('config-cache', context)
     set_state('containerd.restart')
 
 
 @when('containerd.ready')
-@when_any('config.changed.http_proxy', 'config.changed.https_proxy',
+@when_any('containerd.juju-proxy.changed',
+          'config.changed.http_proxy',
+          'config.changed.https_proxy',
           'config.changed.no_proxy')
```

Comment:
```suggestion
@when('containerd.juju-proxy.changed')
```

It seems like `_juju_proxy_changed()` will handle proxy config changes in `config_changed()`.  What do you think about simplifying this function so it only reacts on `containerd.juju-proxy.changed` instead of any of the `.changed.foo` flags?

Without this, `proxy_changed()` may run before `config_changed()` because `.changed.foo` is set, and then run again because `.juju-proxy.changed` gets set in `config_changed()`.